src/target/riscv/riscv-013.c

```diff
@@ -1765,8 +1765,10 @@ static int examine(struct target *target)
 		return ERROR_FAIL;
 	}
 
-	if (!halted)
+	if (!halted) {
 		riscv013_step_or_resume_current_hart(target, false);
+		target->state = TARGET_RUNNING;
```

Comment:
I'd recommend to set here also `target->debug_reason` - to keep the "state" and "debug_reason" in sync. (This is just for consistency. I don't think omission of this would cause any issues.)

```suggestion
		target->state = TARGET_RUNNING;
		target->debug_reason = DBG_REASON_NOTHALTED;		
```